addons/payment/controllers/portal.py

```diff
@@ -114,6 +114,7 @@ def payment_method(self, **kwargs):
             ('payment_flow', '=', 's2s'), ('company_id', '=', request.env.company.id)
         ]))
         partner = request.env.user.partner_id
+        # TODO DBO: check mixed sudoed recordset; use read instead and adapt template?
```

Comment:
not needed, union will preserve the env of the first recordset (so without sudo)
note: weird that here we have access to tokens of the commercial partner, this might need to go for consistency